docs/connectors/sources/aws-s3-file-source.md

```diff
@@ -0,0 +1,175 @@
+# AWS S3 File Source
+
+!!! info
+
+    This is a **Community** connector. Test it before using in production.
+
+    To learn more about differences between Core and Community connectors, see the [Community and Core Connectors](../community-and-core.md) page.
+
+This source reads records from files located in an AWS S3 bucket path and produces 
+them as messages to a kafka topic using any desired `StreamingDataFrame`-based transformations. 
+
+The resulting messages can be produced in "replay" mode, where the time between record 
+producing is matched as close as possible to the original. (per topic partition only).
+
+
+## How To Install
+
+Install Quix Streams with the following optional dependencies:
+
+```bash
+pip install quixstreams[aws]
+```
+
+## How It Works
+
+`FileSource` steps through each folder within the provided path and dumps each record 
+contained in each file as a message to a Kafka topic. Folders are navigated in 
+lexicographical order.
+
+Records are read in a streaming fashion and committed after every file, offering 
+[at-least-once guarantees](#processingdelivery-guarantees).
+
+It can handle one given file type (ex. JSONlines or Parquet) at a time, and also 
+supports file decompression.
+
+You can learn more details about the [expected kafka message format](#message-data-formatschema) below.
+
+## How To Use
+
+S3 File Source is just a special configuration of the `FileSource` connector.
+
+Simply provide it an `S3FileOrigin` (`FileSource(origin=<ORIGIN>)`).
+
+Then, hand the configured `FileSource` to your SDF (`app.dataframe(source=<SOURCE>)`).
+
+For more details around various settings, see [configuration](#configuration).
+
+```python
+from quixstreams import Application
+from quixstreams.sources.community.file import FileSource
+from quixstreams.sources.community.file.origins import S3FileOrigin
+
+app = Application(broker_address="localhost:9092", auto_offset_reset="earliest")
+
+file_origin = S3FileOrigin(
+    aws_s3_bucket="<YOUR BUCKET>",
+    aws_access_key_id="<YOUR KEY ID>",
+    aws_secret_access_key="<YOUR SECRET KEY>",
+    aws_region="<YOUR REGION>",
+)
+source = FileSource(
+    filepath="path/to/your/topic_folder/",
+    file_origin=file_origin,
+    file_format="json",
+    file_compression="gzip",
+)
+sdf = app.dataframe(source=source).print(metadata=True)
+# YOUR LOGIC HERE!
+
+if __name__ == "__main__":
+    app.run()
+```
+
+## Configuration
+
+Here are some important configurations to be aware of (see [File Source API](../../api-reference/sources.md#filesource) for all parameters).
+
+### Required:
+
+`S3FileOrigin`:
+
+- `aws_s3_bucket`: The S3 URI with bucket name only (ex: 's3://your-bucket').
```

Comment:
Lets lose the `s3://` prefix. User should pass only the bucket name.

Reply:
Didn't actually need it, I was just a dum dum and copied it from somewhere haha